bins/stackablectl/src/cmds/stacklets.rs

```diff
@@ -142,80 +133,81 @@ async fn list_cmd(args: &StackletListArgs, common_args: &Cli) -> Result<String,
     }
 }
 
-/// Processes conditions for a single stacklet / product. It returns a
+/// Renders conditions for a single stacklet / product. It returns a
 /// concatenated string of conditions (which are colored green and red) to
 /// display next to each listed stacklet in the table. Additionally, it also
 /// returns a list of errors to be displayed underneath the stacklet table.
-fn process_conditions(
+fn render_conditions(
     product_conditions: Vec<DisplayCondition>,
-    product_index: &mut usize,
+    error_index: &mut usize,
     use_color: bool,
 ) -> (String, Vec<String>) {
     let mut conditions = Vec::new();
     let mut errors = Vec::new();
 
-    let mut condition_index = 0;
     for cond in product_conditions {
-        match process_condition_error(cond.message, cond.is_good, &mut condition_index, use_color) {
-            Some(error) => {
-                errors.push(error);
-                *product_index += 1;
-            }
-            None => (),
-        }
-
         conditions.push(color_condition(
             &cond.condition,
             cond.is_good,
-            *product_index,
-            condition_index,
+            *error_index,
             use_color,
-        ))
+        ));
+
+        match render_condition_error(cond.message, cond.is_good, *error_index, use_color) {
+            Some(error) => {
+                errors.push(error);
+                *error_index += 1;
+            }
+            None => (),
+        };
     }
 
     (conditions.join(", "), errors)
 }
 
-/// Processes one condition and determines if it is an error (not good). If this
+/// Renders one condition and determines if it is an error (not good). If this
 /// is the case, it get colored red and is returned.
-fn process_condition_error(
+fn render_condition_error(
     message: Option<String>,
     is_good: Option<bool>,
-    condition_index: &mut usize,
+    error_index: usize,
     use_color: bool,
 ) -> Option<String> {
     let message = message.unwrap_or("-".into());
 
     match (is_good, use_color) {
-        (Some(false), true) => {
-            *condition_index += 1;
-
-            Some(
-                Red.paint(format!("[{}]: {}", condition_index, message))
-                    .to_string(),
-            )
-        }
+        (Some(false), true) => Some(
+            Red.paint(format!("[{}]: {}", error_index, message))
+                .to_string(),
+        ),
         _ => None,
     }
 }
```

Comment:
Hold on, doesn't this mean that we don't render the message at all if colour is disabled? How about:

```rust
if !is_good.unwrap_or(true) {
    *condition_index += 1;
    let mut message = format!("[{condition_index}]: {message}");
    if use_color {
        message = Red.paint(message).to_string();
    }
    Some(message)
} else {
    None
}
```

Reply:
You are absolutely correct, fixed in bf5960b